### ISSUE TITLE
Support for complex thresholds

### ISSUE THREAD

**Issue by georgemccabe** [dtcenter/METplus#230]
Currently the threshold list variables ([FCST/OBS]_VAR<n>_THRESH) require that the value be a comma-separated list of numbers with a comparison operator in front, i.e. gt3, ==1.2, <=4. Based on a couple use cases, we should support complex thresholds, such as gt3&&le4. How do we want to format them?

Users can get around the need for different values in thresh_lo and thresh_hi for probabilistic data by setting one threshold in the thresh list and then adding in an option for the other, i.e.:

FCST_VAR1_THRESH = gt4
FCAR_VAR1_OPTIONS = thresh_hi = 3

**Comment by georgemccabe** [comment]
Related to #142

**Comment by georgemccabe** [comment]
These changes are made, tested, and merged into develop. The regex for reading thresholds was improved to allow complex thresholds and make sure no unexpected characters are found. I replaced the utilities to pull out the comparison operator and number from a threshold with a function that parses a threshold and creates a list of tuples that contain the comparison operator and number. Updated tests to reflect these changes and added new tests to make sure complex thresholds are supported.
